### PR TITLE
GH-781: New Testnet Polygon Amoy

### PR DESCRIPTION
**Pull request by bertllll**

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ fdlimit = "0.2.1"
 flexi_logger = { version = "0.15.12", features = [ "ziplogs" ] }
 futures = "0.1.31"
 heck = "0.3.3"
-http = "0.2.5"
+http = "0.1.21"
```
Comment (bertllll) [inline node/Cargo.toml]: Hmm. I should revert this.

File: masq_lib/src/test_utils/utils.rs [hunk 2]
```diff
@@ -30,8 +30,16 @@ pub fn ensure_node_home_directory_exists(module: &str, name: &str) -> PathBuf {
 }
 
 pub fn is_running_under_github_actions() -> bool {
-    if let Ok(value) = std::env::var("GITHUB_ACTIONS") {
-        &value == "true"
+    is_env_variable_set("GITHUB_ACTIONS", "true")
+}
+
+pub fn is_test_generated_data_allowed_to_escape_project_dir() -> bool {
+    is_env_variable_set("ALLOW_TEST_DATA_ESCAPE_PROJECT_DIR", "true")
```
Comment (utkarshg6) [inline masq_lib/src/test_utils/utils.rs:37]: Let's pour some more ideas if we can find a way around this.

File: node/Cargo.toml (hunk 1 above)
Comment (utkarshg6) [reply node/Cargo.toml]: Yeah you should.

File: node/src/daemon/setup_reporter.rs [hunk 3]
```diff
@@ -2048,19 +2050,16 @@ mod tests {
     #[test]
     fn get_modified_setup_tilde_in_config_file_path() {
```
Comment (utkarshg6) [inline node/src/daemon/setup_reporter.rs]: I'm glad that you discovered this. Just my personal opinion, I'd be afraid of an application creating files in my computer outside the project directory.

A possible solution: Maybe use container provided by Docker.

**Review by utkarshg6 — CHANGES_REQUESTED** [review]

File: node/src/node_configurator/node_configurator_standard.rs [hunk 4]
```diff
@@ -1204,15 +1207,13 @@ mod tests {
         running_test();
         let _guard = EnvironmentGuard::new();
         let _clap_guard = ClapGuard::new();
-        let home_dir = ensure_node_home_directory_exists(
+        let node_home_dir = ensure_node_home_directory_exists(
```
Comment (utkarshg6) [inline node/src/node_configurator/node_configurator_standard.rs]: I'd prefer `home_dir`. Also, let's change the function name to also say, `ensure_home_directory_exists`.

File: node/src/test_utils/mod.rs [hunk 5]
```diff
@@ -529,6 +535,35 @@ pub struct TestRawTransaction {
     pub data: Vec<u8>,
 }
 
+pub fn make_node_base_dir_and_return_its_absolute_and_relative_path_to_os_home_dir(
```
Comment (utkarshg6) [inline node/src/test_utils/mod.rs]: You mentioned that "why it still exists?". I think you'd like to remove it and change the code inside tests if it's used there. Good Luck :+1:

**Review by utkarshg6 — APPROVED** [review]

**Pull request merged**